test/test_grad_loss.jl

```diff
@@ -0,0 +1,233 @@
+
+function test_grad_discreteAdjoint()
+
+    rgi_ids = ["RGI60-11.03638"]
+    rgi_paths = get_rgi_paths()
+
+    working_dir = joinpath(ODINN.root_dir, "test/data")
+
+    δt = 1/12
+
+    params = Parameters(
+        simulation = SimulationParameters(
+            working_dir=working_dir,
+            use_MB=false,
+            velocities=true,
+            tspan=(2010.0, 2015.0),
+            step=δt,
+            multiprocessing=false,
+            workers=1,
+            light=false, # for now we do the simulation like this (a better name would be dense)
+            test_mode=true,
+            rgi_paths=rgi_paths),
+        hyper = Hyperparameters(
+            batch_size=length(rgi_ids), # We set batch size equals all datasize so we test gradient
+            epochs=100,
+            optimizer=ODINN.ADAM(0.005)),
+            # optimizer=ODINN.Descent(0.001)),
+        UDE = UDEparameters(
+            sensealg=ODINN.ZygoteAdjoint(),
+            optim_autoAD=ODINN.NoAD(),
+            grad=ODINNDiscreteAdjoint(),
+            optimization_method="AD+AD",
+            target = "A"),
+        solver = Huginn.SolverParameters(
+            step=δt,
+            save_everystep=true,
+            progress=true)
+    )
+
+    model = Model(
+        iceflow = SIA2Dmodel(params),
+        mass_balance = TImodel1(params; DDF=6.0/1000.0, acc_factor=1.2/1000.0),
+        machine_learning = NeuralNetwork(params))
+
+    # We retrieve some glaciers for the simulation
+    glaciers = initialize_glaciers(rgi_ids, params)
+
+    # Time stanpshots for transient inversion
+    tstops = collect(2010:δt:2015)
+
+
+    ### Fake law for A(T) for Peterson & Cuffey
```

Comment:
Since this part is common to both tests, maybe we could externalize it from the functions?